### PR TITLE
Add French translation

### PR DESCRIPTION
**Pull request by arkhi**
Hello…

I’m not entirely happy with things like “URL pour soutenir l’instance” but unless there is a paragraph to describe what the context of “support” is, I found it hard to make it shorter.

Also… "Texte affiché lorsque les enregistrements sont clos :" is a bit mouthful but, for now, I take it as a good way to see if the UI is solid enough. :)

I used “Comptes” (_accounts_) for _users_ as it is more gender‑neutral. It would be _utilisateurs et utilisatrices_ otherwise. Mastodon is doing the same.

I didn’t test thoroughly but it should be a fairly safe start already. I think the following could cause some issues:

- Strings like `in a <a href=\"%(related_path)s\">%(preview_name)s</a>` will be problematic with gender‑based languages like French, Spanish and many others. _a_ can be _un_ or _une_ depending on the gender of the object.
- "Joined %(date)s" has _ago_ included at the end of the variable. In french the duration (not a date, if I understood what I saw right) listed should happen after _il y a_: _Joined 2 days ago_ becomes _Enregistré il y a deux jours_ (deux = 2).

Let me know. :)

### CODE REVIEW

**Comment by arkhi** [comment]
By the way, I added the file manually. Running the command in the README gave me the following:

```sh
me@computer:~/Projects/bookwyrm$ ./bw-dev makemessages -l fr
+ case "$CMD" in
+ django-admin makemessages --extension html --ignore=venv3 -l fr
./bw-dev: line 94: django-admin: command not found
```

File: locale/fr_FR/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -0,0 +1,1446 @@
...
+msgstr "Modifier l’auteur ou autrice"
+
+#: bookwyrm/templates/author.html:29
+msgid "Wikipedia"
+msgstr "Wikipedia"
+
+#: bookwyrm/templates/author.html:34
+#, python-format
+msgid "Books by %(name)s"
+msgstr Livres par %(name)s
```
Comment (mouse-reeve) [inline locale/fr_FR/LC_MESSAGES/django.po:32]: this is missing quotes

**Comment by mouse-reeve** [comment]
> By the way, I added the file manually. Running the command in the README gave me the following:
> 
> ```shell
> me@computer:~/Projects/bookwyrm$ ./bw-dev makemessages -l fr
> + case "$CMD" in
> + django-admin makemessages --extension html --ignore=venv3 -l fr
> ./bw-dev: line 94: django-admin: command not found
> ```

hmm, it works for me, so I wonder if there's a configuration step I failed to document to make the `django-admin` tool work 🤔

**Comment by mouse-reeve** [comment]
Speaking as someone who knows zero French, this looks great! I'm going to go through the UI and improve the spots you pointed out, and now that there's a translation in place, it's much easier to see the things in the UI I missed. 

My only requested change is the missing quotes :)

I really appreciate how promptly and thoroughly you did this! Thank you!

**Comment by arkhi** [comment]
Ha, damn! I searched for an easily accessible gettext linter and did not find anything that worked or that wouldn’t take me less than 5 minutes to install.

I’ll update that tomorrow.

File: locale/fr_FR/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -0,0 +1,1446 @@
...
+msgstr "Activité"
+
+#: bookwyrm/templates/user/user_layout.html:54
+msgid "Reading Goal"
+msgstr "Défi lecture"
+
+#: bookwyrm/templates/user/user_preview.html:13
+#, python-format
+msgid "Joined %(date)s"
+msgstr "Enregistré(e) il y a %(date)s"
```
Comment (mouse-reeve) [inline locale/fr_FR/LC_MESSAGES/django.po:1446]: it looks like the built in django date formatter is adding `il y a` on its own, so I think just `"Enregistré(e) %(date)s"` will do the trick here?

**Comment by mouse-reeve** [comment]
okay I went through and fixed the obvious places I missed template translations, changes are in #666. I didn't find a super obvious solution to why the django-admin command failed, it sounds like it might be platform dependent? (https://stackoverflow.com/questions/8250086/command-not-found-django-admin-py)

> Strings like in a <a href=\"%(related_path)s\">%(preview_name)s</a> will be problematic with gender‑based languages like French, Spanish and many others. a can be un or une depending on the gender of the object.

I'm not sure that there's a good template-level fix for this, since that texts is preceded by a username, so there's no way to know the gender of the subject. All my support for the English singular 'they" isn't much use in a language with grammatical gender :)

If getting the `django-admin makemessages --extension html --ignore=venv3 -l fr` command working is a hassle, I can run the command over the translation file you created, since manually merging the new changes would be wildly hard.

File: locale/fr_FR/LC_MESSAGES/django.po (hunk 1 above)
Comment (arkhi) [reply locale/fr_FR/LC_MESSAGES/django.po]: This is fixed in https://github.com/mouse-reeve/bookwyrm/pull/665/commits/6e9fc0dbba3ec53274c7a90d202234b204bb9f3f.

File: locale/fr_FR/LC_MESSAGES/django.po (hunk 2 above)
Comment (arkhi) [reply locale/fr_FR/LC_MESSAGES/django.po]: This is fixed in https://github.com/mouse-reeve/bookwyrm/pull/665/commits/4a178d05cda54c70bac7f46a877bbbe1ab8262da.

**Comment by arkhi** [comment]
> okay I went through and fixed the obvious places I missed template translations, changes are in #666. I didn't find a super obvious solution to why the django-admin command failed, it sounds like it might be platform dependent? (https://stackoverflow.com/questions/8250086/command-not-found-django-admin-py)

A very wild guess tells me you’re on Windows. I’m on Linux. I don’t know what would happen on Mac; I would suspect something closer to Linux than Windows.

> > Strings like `in a <a href=\"%(related_path)s\">%(preview_name)s</a>` will be problematic with gender‑based languages like French, Spanish and many others. a can be un or une depending on the gender of the object.
> 
> I'm not sure that there's a good template-level fix for this, since that texts is preceded by a username, so there's no way to know the gender of the subject. All my support for the English singular 'they" isn't much use in a language with grammatical gender :)

The problem is more about the object than the subject in that case. In the context of the example I gave:

- _John Doe mentioned you in a &lt;a…&gt;**Review**&lt;/a&gt;_
  _John Doe a mentionné votre nom dans **une &lt;a…&gt;critique&lt;/a&gt;**_
- _John Doe mentioned you in a &lt;a…&gt;**Comment**&lt;/a&gt;_
  _John Doe a mentionné votre nom dans **un &lt;a…&gt;commentaire&lt;/a&gt;**_

> If getting the `django-admin makemessages --extension html --ignore=venv3 -l fr` command working is a hassle, I can run the command over the translation file you created, since manually merging the new changes would be wildly hard.

Yes please; I have no habits working with Docker or Python, so debugging the environment would take me a fair amount of time.

From what I checked, there is no sign of django-admin being installed when following the [steps in the README](https://github.com/mouse-reeve/bookwyrm#setting-up-the-developer-environment).

**Comment by mouse-reeve** [comment]
right! object not subject, that makes much more sense.  I'll try and fix those

**Comment by mouse-reeve** [comment]
I think #667 will fix the problems with `makemessages`

**Comment by mouse-reeve** [comment]
Github doesn't supporting uploading .po files (probably a reasonable call), but I can send you the version updated to reflect #666 through whatever means works best for you if the `makemessages` command is still janky for you

**Comment by arkhi** [comment]
> I think #667 will fix the problems with `makemessages`

That worked indeed!

The last commit has updated strings. I will wait for this PR to be merged before continuing with the strings you added in #666 though, unless this is a very big issue for you.

**Pull request merged**